### PR TITLE
feature: stripe subscriptions

### PR DESCRIPTION
**Pull request by BarryThePenguin**

### CODE REVIEW

File: src/components/account/manage-account-form.tsx [hunk 1]
```diff
@@ -148,50 +110,10 @@ const ManageForm: FC<ManageFormProps> = ({hasFreeAccount, hasPaidAccount}) => {
...
-                type="password"
-                label="Confirm password"
-                name="confirmPassword"
-              />
-            </Box>
-            <Box marginBottom={3}>
-              <Button type="submit">Change Password</Button>
-            </Box>
-          </Form>
-        </Formik>
```
Comment (BarryThePenguin) [inline src/components/account/manage-account-form.tsx]: We need to rethink changing passwords. Still need to read the auth0 docs https://auth0.com/docs/connections/database/password-change

File: lib/graphql/models/user.ts [hunk 2]
```diff
@@ -18,42 +22,70 @@ export async function getById(id: string): Promise<User> {
...
+
+  await sanity
+    .patch(_id)
+    .set({
+      stripeCustomerId,
+      invoiceStatus,
+      periodEndDate: periodEndDate.toISOString()
+    })
+    .commit();
+}
```
Comment (BarryThePenguin) [inline lib/graphql/models/user.ts]: This is where we send the Stripe details to Sanity

I tried doing this in one operation, but couldn't figure out how to do a `patch()` with a query _and_ query variables..

@readeral can update the `invoiceStatus` field to have the status values please?

File: lib/graphql/models/user.ts [hunk 3]
```diff
@@ -18,42 +22,70 @@ export async function getById(id: string): Promise<User> {
...
       _id,
       name,
       email,
+      invoiceStatus,
+      periodEndDate,
+      stripeCustomerId,
       "role": permission.role,
       shortlist[]->{_id,_type,title,hymnNumber}
   }`,
-    {googleProviderId: user.sub}
```
Comment (BarryThePenguin) [inline lib/graphql/models/user.ts]: @readeral I think we can ditch `googleProviderId` field in Sanity now.. It's easier if we just use their `email` instead

That way people can log into auth0 with either email/password, or Google login

Comment (BarryThePenguin) [reply lib/graphql/models/user.ts]: Oh, and we can remove the "Profile photo" field now too

File: lib/graphql/models/user.ts (hunk 2 above)
Comment (readeral) [reply lib/graphql/models/user.ts]: updated

I'll look into chaining fetch and patch more

File: src/components/account/manage-account-form.tsx (hunk 1 above)
Comment (readeral) [reply src/components/account/manage-account-form.tsx]:
```
var request = require("request");

var options = {
  method: 'PATCH',
  url: 'https://YOUR_DOMAIN/api/v2/users/USER_ID',
  headers: {'content-type': 'application/json'},
  body: {password: 'NEW_PASSWORD', connection: 'Username-Password-Authentication'},
  json: true
};

request(options, function (error, response, body) {
  if (error) throw new Error(error);

  console.log(body);
});
```

Comment (readeral) [reply src/components/account/manage-account-form.tsx]: https://auth0.com/docs/connections/database/password-change#use-the-management-api

**Pull request merged**